### PR TITLE
Prevent use of SU advection with discontinuous function spaces

### PR DESCRIPTION
**Pull request by Patol75**
Closes gh-62

Pretty much self-explanatory; see the linked issue for additional detail.

### CODE REVIEW

**Review by sghelichkhani — APPROVED** [review]
Looks good to me!

**Comment by sghelichkhani** [comment]
I'll wait for Will to approve before merging! But unlikely that there would be anything wrong.

**Review by w3168 — APPROVED** [review]
Thanks Thomas! Looks good!

**Pull request merged**
